src/model/window-guardian.ts

```diff
@@ -67,23 +67,46 @@ const makeWindowGuardianConfig = (
 };
 
 export const makeGuardianBrowserCAPI = (CAPI: CAPIType): CAPIBrowserType => {
-    // it is important to pass down the index of rich links as well as the component itself
-    const richLinksWithIndex: RichLinkBlockElement[] = CAPI.blocks[0].elements.reduce(
-        (acc, element, index: number) => {
+    // filter out RichLinkBlockElement from elements
+    const richLinks: RichLinkBlockElementBrowser[] = CAPI.blocks[0].elements.reduce(
+        (acc, element, index) => {
             if (
                 element._type ===
                 'model.dotcomrendering.pageElements.RichLinkBlockElement'
             ) {
                 acc.push({
-                    ...element,
+                    url: element.url,
+                    // It is important to pass down the index of rich links as well as the component itself
                     richLinkIndex: index,
-                } as RichLinkBlockElement);
+                } as RichLinkBlockElementBrowser);
             }
             return acc;
         },
-        [] as RichLinkBlockElement[],
+        [] as RichLinkBlockElementBrowser[],
     );
 
+    // references with the key `rich-link` work similar to RichLinkBlockElement
+    // however we do care about thier index, as we will use spaceFinder to determin thier location
```

Comment:
```suggestion
    // however we do care about their index, as we will use spaceFinder to determine their location
```

Reply:
is this supposed to be `don't care about their index` rather than `do`? Otherwise the code doesn't match the comment :)